service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/InputBundle.java

```diff
@@ -24,4 +32,21 @@ public InputBundle(Bundle bundle) {
     public Optional<ListResource> getListReferencedToEncounter(IIdType reference, String code) {
         return extractListByEncounterReference(this.bundle, reference, code);
     }
+
+    public List<Condition> getRelatedConditions(String referenceId) {
+        return bundle.getEntry()
+            .stream()
+            .filter(Bundle.BundleEntryComponent::hasResource)
+            .map(Bundle.BundleEntryComponent::getResource)
+            .filter(resource -> ResourceType.Condition.equals(resource.getResourceType()))
+            .map(resource -> (Condition) resource)
+            .filter(condition -> filterExtensionByUrl(condition, ACTUAL_PROBLEM_URL)
+                    .map(Extension::getValue)
+                    .map(Reference.class::cast)
+                    .map(Reference::getReference)
+                    .stream().findFirst()
+                    .filter(referenceId::equals)
+                    .isPresent())
```

Comment:
The optional -> stream -> optional chain here is unnecessary - sorry if that's come from my earlier suggestion!

```suggestion
                    .map(Reference::getReference)
                    .filter(referenceId::equals)
                    .isPresent())
```

Reply:
True! Thanks!